engine/src/components/component_utils.cpp

```diff
@@ -135,6 +137,23 @@ EnergyContainer* GetSourceFromConfiguration(const EnergyConsumerSource source, E
     }
 }
 
+EnergyContainer* GetSourceFromConfiguration(const std::string& energy_source, EnergyContainer* fuel,
+    EnergyContainer* energy, EnergyContainer* ftl_energy)
+{
+    if (energy_source == "Infinite" || energy_source == "infinite") {
```

Comment:
Boost includes a handy algorithm for case insensitive. 

```
#include <boost/algorithm/string.hpp>
// Or, for fewer header dependencies:
//#include <boost/algorithm/string/predicate.hpp>

std::string str1 = "hello, world!";
std::string str2 = "HELLO, WORLD!";

if (boost::iequals(str1, str2))
{
    // Strings are identical
}

```
Source: https://stackoverflow.com/a/315463/5276890

Reply:
That's right, I remember seeing that before. Thanks for the suggestion. Maybe I can apply it in the next pass.